src/util/tools.cc

```diff
@@ -1205,6 +1205,8 @@ int find_local_port(in_port_t range_min, in_port_t range_max)
         retry_count++;
     } while (retry_count < std::numeric_limits<uint16_t>::max());
 
+    close(fd);
```

Comment:
In my opinion it can / should be placed in the end of the while loop if connect fails.

Reply:
Sounds right.